tests/integration/preprocessor/_io/test_concatenate.py

```diff
@@ -14,34 +16,174 @@ class TestConcatenate(unittest.TestCase):
 
     def setUp(self):
         """Start tests."""
-        coord = DimCoord([1, 2], var_name='coord')
-        second_coord = coord.copy([3, 4])
-        third_coord = coord.copy([5, 6])
+        self._model_coord = DimCoord(
+            [1., 2.], var_name='time', standard_name='time',
+            units='days since 1950-01-01'
+        )
         self.raw_cubes = []
+        self._add_cube([1., 2.], [1., 2.])
+        self._add_cube([3., 4.], [3., 4.])
+        self._add_cube([5., 6.], [5., 6.])
+
+    def _add_cube(self, data, coord):
         self.raw_cubes.append(
-            Cube([1, 2], var_name='sample', dim_coords_and_dims=((coord,
-                                                                  0), )))
-        self.raw_cubes.append(
-            Cube([3, 4],
-                 var_name='sample',
-                 dim_coords_and_dims=((second_coord, 0), )))
-        self.raw_cubes.append(
-            Cube([5, 6],
+            Cube(data,
                  var_name='sample',
-                 dim_coords_and_dims=((third_coord, 0), )))
+                 dim_coords_and_dims=((self._model_coord.copy(coord), 0), )
+                 )
+        )
 
     def test_concatenate(self):
         """Test concatenation of two cubes."""
         concatenated = _io.concatenate(self.raw_cubes)
-        self.assertTrue((concatenated.coord('coord').points == np.array(
+        self.assertTrue((concatenated.coord('time').points == np.array(
             [1, 2, 3, 4, 5, 6])).all())
 
-    def test_fail_with_duplicates(self):
-        """Test exception raised if two cubes are overlapping."""
-        self.raw_cubes.append(self.raw_cubes[0].copy())
-        with self.assertRaises(ValueError):
+    def test_concatenate_with_overlap(self):
+        """Test concatenation of time overalapping cubes"""
+        self._add_cube([6.5, 7.5], [6., 7.])
+        concatenated = _io.concatenate(self.raw_cubes)
+        self.assertTrue(np.allclose(
+            concatenated.coord('time').points,
+            np.array([1., 2., 3., 4., 5., 6., 7.])
+        ))
+        self.assertTrue(np.allclose(
+            concatenated.data,
+            np.array([1., 2., 3., 4., 5., 6.5, 7.5])
+        ))
+
+    def test_concatenate_with_overlap_2(self):
+        """Test a more generic case."""
+        self._add_cube([65., 75.], [3., 200.])
+        self._add_cube([65., 75.], [1000., 7000.])
+        concatenated = _io.concatenate(self.raw_cubes)
+        self.assertTrue(np.allclose(
+            concatenated.coord('time').points,
+            np.array([1., 2., 3., 4., 5., 6., 1000., 7000.])
```

Comment:
Where did the time point 200 as defined on line 57 go?

Reply:
Thing is, the overlap can not be one single border point, so the [3, 200] cube is not counted in

Reply:
At the very least, I think it would be good to add some warning log messages telling the user that some data is ignored. 

If there is overlapping data, I think it would be nice to have a debug log message telling the user what data from what cube you're using.

Reply:
Yeah that'd be very sensible, but it's a bit of a headache. I'll see what I can do after Xmas :grin: I still think we should get this in sopner than later because it's quite needed, maybe we can open an issue and a new PR for these logger mssges after merge?

Reply:
I would prefer to have it working properly before merging. Silently ignoring data makes the tool really hard to understand.